patches/server/1039-General-ItemMeta-fixes.patch

```diff
@@ -248,6 +296,15 @@ index 8e0dd4b7a7a25a8beb27b507047bc48d8227627c..cf5d27ccc2225bac3aa57912f444f95d
  
          itemTag.put(CraftMetaFirework.FIREWORKS, new Fireworks(this.power, effects));
      }
+@@ -218,7 +231,7 @@ class CraftMetaFirework extends CraftMetaItem implements FireworkMeta {
+     }
+ 
+     boolean isFireworkEmpty() {
+-        return !(this.hasEffects() || this.hasPower());
++        return !(this.effects != null || this.hasPower()); // Paper - empty effects list should stay on the item
```

Comment:
I think by changing this, you also need to change `hasEffects` or at least change in the places where `hasEffects` is used, like equalsCommon, hashCode, serialize